### PR TITLE
Feature/sd

### PR DESCRIPTION
**Pull request by kels271828**
Added optional argument `stdev` to `Smoother.__call__`.

### CODE REVIEW

File: src/weave/smoother.py [hunk 1]
```diff
@@ -249,77 +250,72 @@ def check_args(data: DataFrame, observed: List[str],
...
-
-    def check_data(self, data: DataFrame, observed: List[str],
-                   smoothed: List[str], fit: Optional[str],
+        if stdev is not None:
+            if observed == stdev:
+                raise ValueError('`observed` == `stdev`')
+            if smoothed is not None and stdev == smoothed:
+                raise ValueError('`stdev` == `smoothed`')
+        if smoothed is not None and observed == smoothed:
+            raise ValueError('`observed` == `smoothed`')
```
Comment (zhengp0) [inline src/weave/smoother.py]: Here maybe we can use a `set` to see if there are any overlap between `observed, stdev, smoothed`

File: src/weave/smoother.py [hunk 2]
```diff
@@ -118,27 +118,29 @@ def dimensions(self, dimensions: List[Dimension]) -> None:
 
         self._dimensions = dimensions
 
-    def __call__(self, data: DataFrame, observed: Union[str, List[str]],
-                 smoothed: Optional[Union[str, List[str]]] = None,
+    def __call__(self, data: DataFrame, observed: str,
```
Comment (zhengp0) [inline src/weave/smoother.py:121]: I like `observed` is just one column now, simplify things a lot!

**Review by zhengp0 — APPROVED** [review]
Looks great!

File: src/weave/smoother.py (hunk 2 above)
Comment (kels271828) [reply src/weave/smoother.py:121]: Yes, it made things much easier for validation and the new standard deviation feature :)

File: src/weave/smoother.py (hunk 1 above)
Comment (kels271828) [reply src/weave/smoother.py]: Good idea

**Pull request merged**